src/hooks/useForm.ts

```diff
@@ -0,0 +1,69 @@
+import type { AbstractForm, Status } from '~/src/types'
+import { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+import {
+  getForm,
+  useSelector,
+  getFormError,
+  setFormStatus,
+  getFormStatus,
+  setForm as _setForm,
+  setField as _setField,
+  clearForm as _clearForm,
+} from '~/src/store'
+
+export const useForm = <Form extends AbstractForm>(
+  formId: string,
+  defaultForm: Form
+) => {
+  const dispatch = useDispatch()
+  const form = useSelector((state) => getForm(state, formId, defaultForm))
+  const status = useSelector((state) => getFormStatus(state, formId))
+  const error = useSelector((state) => getFormError(state, formId))
+
+  const setField = useCallback(
+    <Name extends keyof Form & string, Value extends Form[Name]>(
+      name: Name,
+      value: Value
+    ) => {
+      dispatch(_setField({ formId, name, value }))
+    },
+    [dispatch, formId]
+  )
+
+  const setForm = useCallback(
```

Comment:
is it possible to redesign this so it works more like `useState`? e.g.
```ts
setForm((form) => ({ ...form, someField: 'some new value' }))
```

that should simplify things on the component end.